client_example/32_convnd_fwd_convscale/common.hpp

```diff
@@ -0,0 +1,385 @@
+                                                std::multiplies<>());
+}
+
+template <typename WeiDataType, ck::index_t NumDimSpatial, ck::index_t NumNonSpatialDim = 3>
+std::size_t
+GetWeightByte(const std::array<ck::index_t, NumDimSpatial + NumNonSpatialDim>& weights_lengths)
+{
+    // sizeof(WeiDataType) * (G * K * C * <filter spatial lengths product>) +
+    return sizeof(WeiDataType) * std::accumulate(std::begin(weights_lengths),
+                                                 std::end(weights_lengths),
+                                                 static_cast<std::size_t>(1),
+                                                 std::multiplies<>());
+}
+
+template <typename OutDataType, ck::index_t NumDimSpatial, ck::index_t NumNonSpatialDim = 3>
+std::size_t
+GetOutputByte(const std::array<ck::index_t, NumDimSpatial + NumNonSpatialDim>& output_lengths)
+{
+    // sizeof(OutDataType) * (G * N * K * <output spatial lengths product>);
+    return sizeof(OutDataType) * std::accumulate(std::begin(output_lengths),
+                                                 std::end(output_lengths),
+                                                 static_cast<std::size_t>(1),
+                                                 std::multiplies<std::size_t>());
+}
+
+template <ck::index_t NumDimSpatial,
+          typename InDataType,
+          typename WeiDataType,
+          typename D0DataType,
+          typename D1DataType,
+          typename D2DataType,
+          typename OutDataType,
+          typename InLayout,
+          typename WeiLayout,
+          typename D0Layout,
+          typename D1Layout,
+          typename D2Layout,
+          typename OutLayout,
+          ck::index_t NumNonSpatialDim = 3,
+          typename AComputeType        = InDataType,
+          typename BComputeType        = AComputeType>
+bool run_grouped_conv_fwd_convscale(
+    std::array<ck::index_t, NumDimSpatial + NumNonSpatialDim> in_lengths,
+    std::array<ck::index_t, NumDimSpatial + NumNonSpatialDim> wei_lengths,
+    std::array<ck::index_t, NumDimSpatial + NumNonSpatialDim> d0_lengths,
+    std::array<ck::index_t, NumDimSpatial + NumNonSpatialDim> d1_lengths,
+    std::array<ck::index_t, NumDimSpatial + NumNonSpatialDim> d2_lengths,
+    std::array<ck::index_t, NumDimSpatial + NumNonSpatialDim> out_lengths)
+{
+    std::size_t in_mem_size  = GetInputByte<InDataType, NumDimSpatial>(in_lengths);
+    std::size_t wei_mem_size = GetWeightByte<WeiDataType, NumDimSpatial>(wei_lengths);
+    std::size_t d0_mem_size  = GetOutputByte<D0DataType, NumDimSpatial>(d0_lengths);
+    std::size_t d1_mem_size  = GetOutputByte<D1DataType, NumDimSpatial>(d1_lengths);
+    std::size_t d2_mem_size  = GetOutputByte<D2DataType, NumDimSpatial>(d2_lengths);
+    std::size_t out_mem_size = GetOutputByte<OutDataType, NumDimSpatial>(out_lengths);
+
+    SimpleDeviceMem in(in_mem_size);
+    SimpleDeviceMem wei(wei_mem_size);
+    SimpleDeviceMem d0(d0_mem_size);
```

Comment:
It should be enough to allocate memory only for 1 element and pass {N, Do, Ho, Wo, G, K} lengths and {0, 0,0, 0, 0, 0} strides for broadcast.